### PR TITLE
Remove deprecated `CertAuthTypeAll`

### PR DESCRIPTION
**Pull request by rosstimothy**

### CODE REVIEW

**Review by smallinsky — APPROVED** [review]
LGTM.
Could we also align our docs and get rid of reference to `--type=all` like in: https://github.com/gravitational/teleport/blob/757a80e75fcf0836c6fda94d7842938f46d46cb8/docs/pages/management/operations/ca-rotation.mdx#L90

File: tool/tctl/common/auth_command.go [hunk 1]
```diff
@@ -363,10 +363,7 @@ func (a *AuthCommand) generateSnowflakeKey(ctx context.Context, clusterAPI auth.
 // RotateCertAuthority starts or restarts certificate authority rotation process
 func (a *AuthCommand) RotateCertAuthority(ctx context.Context, client auth.ClientI) error {
 	if a.rotateType == "" {
-		return trace.BadParameter("required flag --type not provided; previous versions defaulted to --type=all which is deprecated and will be removed in a future version")
-	}
-	if a.rotateType == string(types.CertAuthTypeAll) {
-		fmt.Println("\033[0;31mNOTICE:\033[0m --type=all will be deprecated in a future version")
+		return trace.BadParameter("required flag --type not provided")
```
Comment (zmb3) [inline tool/tctl/common/auth_command.go:366]: Why not just mark the flag required with kingpin and let kingpin handle this check?

**Review by espadolini — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
Tagged with `release-notes` so we make sure to include a note in the v13 release notes.

**Comment by GavinFrazar** [comment]
@zmb3 @rosstimothy @jakule 

This PR removed `findDuplicatedCertificates`, which was introduced in https://github.com/gravitational/teleport/pull/9593/files#diff-511d4992b0f9dfe420da397fcbd636009b705e58dc53893877ee78bef0b6e10eR432 to handle the DatabaseCA migration.

The DatabaseCA migration path v9->v10 created the DatabaseCA as a copy of the HostCA, so DatabaseCA == HostCA initially. 

On the first rotation of either CA, we wanted them both to be rotated.

We had this duplicate finder func to prevent a scenario where a customer rotates one of HostCA or DatabaseCA and inadvertently leaves a CA behind that still uses the old key. See this discussion: https://github.com/gravitational/teleport/pull/9593#discussion_r782689579

If any clusters migrated v9->v10 and did not rotate their CAs since, then they will unknowingly run into the scenario described above. Looks like this was released in v13.0.0. I think we should advise customers of this.
I'd also like to re-add the `findDuplicatedCertificates` func.

**Comment by espadolini** [comment]
Wouldn't rotating the DB CA break access to all the on-prem databases at once? That's 100% not something we should ever do unless a cluster admin deliberately asks for it.

**Comment by GavinFrazar** [comment]
The on-prem databases were configured to trust the HostCA before the DatabaseCA was created, so rotating the HostCA necessarily breaks database access regardless of whether we rotate both.

**Pull request merged**